src/main.rs

```diff
@@ -33,12 +33,22 @@ impl Service for HelloWorld {
     type Future = Box<Future<Item=Self::Response, Error=Self::Error>>;
 
     fn call(&self, req: Self::Request) -> Self::Future {
-        // TODO: parse body.
-        let path = String::from(req.uri().path());
+
         let tx = self.sender.clone();
-        Box::new(tx.send(path)
-            .map_err(|error| {
-                println!("Could not send path: {}", error);
+        let work =req.body()
```

Comment:
You could use rustfmt: `rustup component add --toolchain nightly rustfmt-preview` ;)